### ISSUE TITLE
Add the KNN Model

### ISSUE THREAD

**Issue by Soumya-Kushwaha** [Rakesh9100/ML-Project-Drug-Review-Dataset#11]
### Prerequisites

- [X] I am an Open Source Contributor
- [X] I checked to make sure that this issue has not already been filed

### Description

Add alternative models such as SVM, KNN etc. and try results

### Code of Conduct

- [X] I follow [Contributing Guidelines](https://github.com/Rakesh9100/ML-Project-Drug-Review-Dataset/blob/main/.github/CONTRIBUTING_GUIDELINES.md) of this project.

**Comment by lcs2022026** [comment]
@Rakesh9100  
i am GSSOC'23 contributor
i would like to work on this project 
the dataset i am gonna use is iris-flower dataset from sklearn.datasets
@Rakesh9100

**Comment by Vethan123** [comment]
Hii @Rakesh9100 ,
i am GSSOC'23 contributor
i am willing to work on this issue
i gonna work on boston house pricing dataset and also on titanic dataset for different models to check their accuracy.
so please assign this to me

**Comment by Saiesh-Agre** [comment]
Hello @Soumya-Kushwaha 
I would like to contribute to this issue. I just finished working for Fake News classification. 
Since the project concerns with Drug classification, I would use this Kaggle dataset (https://www.kaggle.com/datasets/prathamtripathi/drug-classification). The Dataset would be effective in evaluating the results.
I will also check for generalization of the model.

It would be great if you would assign it to me.

**Comment by tushtithakur** [comment]
Hi, I'm excited about the ML Project Drug Review Dataset's aim to analyze drug effectiveness using patient reviews. I have experience in developing ML models for similar tasks and would like to explore alternative models (e.g., KNN, Decision Tree Classifier, Random Forest) on the dataset. 
I would be grateful if you could assign this task to me. Looking forward to collaborating.

**Comment by UditSharma9999** [comment]
Hi i am kaggle expert. I have experience in developing ML models for similar tasks . 
I want to work on this issue. Please assign me.

**Comment by Veena4512** [comment]
Hello @Rakesh9100 I love to solve this issue. I have worked on different machine learning models . Please assign this to me.

**Comment by sijalalyy** [comment]
I am GSSOC 23 contributor assign this to me i want to work on this

**Comment by Rakesh9100** [comment]
@Soumya-Kushwaha Assigning you, Go ahead
Remember similar issues are already been opened before but still if you want you can try, the one which will be good will be merged.

**Comment by khushi767** [comment]
Could you please assign me this as a GSSOC Contributor.

**Comment by Mubashir-Ul-Hassan** [comment]
Hi, @Rakesh9100 I would like to contribute. Please assign the task to me under GSSOC'23.

**Comment by Rakesh9100** [comment]
@Soumya-Kushwaha Provide the update pls

**Comment by Soumya-Kushwaha** [comment]
@Rakesh9100 I'll make the PR today!

**Comment by Rakesh9100** [comment]
@tushtithakur Do you want to work

**Comment by tushtithakur** [comment]
@Rakesh9100 Sure..I would like to contribute..

**Comment by Rakesh9100** [comment]
Pls tell in brief the work you will do @tushtithakur

**Comment by tushtithakur** [comment]
@Rakesh9100 I would apply SVM,KNN,Random tree etc algorithms on the dataset and get results out of it.

**Comment by tushtithakur** [comment]
@Rakesh9100 Thank you for considering me for exploring alternative machine learning models on the dataset. I am eager to incorporate models like SVM, KNN, Decision Tree Classifier, and Random Forest. I will evaluate their performance using accuracy, precision, etc. Through careful analysis, I will provide insights into the strengths and weaknesses of each model.

**Comment by Rakesh9100** [comment]
The algorithms you specifies are already been applied @tushtithakur

**Comment by tushtithakur** [comment]
@Rakesh9100, thank you for your suggestion. Building upon the existing models applied, I will further expand the analysis by incorporating additional machine-learning models into the code. In addition to the previously mentioned models like SVM, KNN, Decision Tree Classifier, and Random Forest, I will include Gradient Boosting, XGBoost, LightGBM, Naive Bayes, AdaBoost, and SVM. By incorporating these various algorithms, we can explore their strengths and weaknesses in the context of the dataset.

**Comment by Rakesh9100** [comment]
Okay pls proceed @tushtithakur

**Comment by Rakesh9100** [comment]
Update pls @tushtithakur

**Comment by Pradhyumna789** [comment]
Hi, @Rakesh9100 I would like to contribute. Please assign the task to me under GSSOC'23.

**Comment by Pradhyumna789** [comment]
Right now I'm working on the CIFAR10 dataset using KNN, SVM, and PCA. I think I'll be able to solve this issue.

**Comment by tushtithakur** [comment]
@Rakesh9100 I'll make the PR by tomorrow.

**Comment by Rakesh9100** [comment]
@Pradhyumna789 Do you want to be assigned?

**Comment by karthikbhandary2** [comment]
can I do this? @Rakesh9100

**Comment by Rakesh9100** [comment]
Pls complete your assigned issue first and then will assign you this @karthikbhandary2

**Comment by karthikbhandary2** [comment]
ok @Rakesh9100

**Comment by Rakesh9100** [comment]
Do you want to be assigned? @karthikbhandary2

**Comment by karthikbhandary2** [comment]
yes @Rakesh9100

**Comment by karthikbhandary2** [comment]
Thanks for the assign!

**Comment by Rakesh9100** [comment]
Update of your work @karthikbhandary2

**Comment by karthikbhandary2** [comment]
Almost done @Rakesh9100 I will be creating a PR shortly!
